errors/list.go

```diff
@@ -150,6 +150,15 @@ func (l *List) Append(errs ...error) {
 	}
 }
 
+// AppendWrap is like Append() but wrap all errors with the provided kind.
+func (l *List) AppendWrap(kind Kind, errs ...error) {
```

Comment:
Hmmm maybe call it AppenKind and document that it adds the kind to all errors ? Wrapping seems misleading because we are not doing error wrapping in the sense that calling Unwrap will not return the Kind (the kind is not even an error, etc), so we are not wrapping up an error inside another. You could argue that you are wrapping the Kind field inside the error, and that is not necessarily wrong, but on Go idioms it seems misleading since wrapping is used about errors. But Im not sure, feel free to disregard.

Reply:
this is the exact wording we used when we designed the `errors` package...
https://github.com/mineiros-io/terramate/blob/main/errors/error.go#L182
when we call `errors.E(arg1, arg2, errlist)` we "wrap" all errors with the arguments provided to `E`.
It's not exactly the same meaning as Go's wrap (even though we support also "wrap" errlist in the Go sense, because we store errlist in the Error.Err which we return in the `Unwrap()` method.
So I think the "word" wrap applies and it's the actual concept we use, it's strange to avoid a word in this case.

I don't like `AppendKind()` because then it looks like we appending a kind to the errlist... which is not the case here.